python/ql/src/Classes/CallsToInitDel/MethodCallOrder.qll

```diff
@@ -0,0 +1,221 @@
+/** Definitions for reasoning about multiple or missing calls to superclass methods. */
+
+import python
+import semmle.python.ApiGraphs
+import semmle.python.dataflow.new.internal.DataFlowDispatch
+import codeql.util.Option
+
+/** Holds if `meth` is a method named `name` that transitively calls `calledMulti` of the same name via the calls `call1` and `call2`. */
+predicate multipleCallsToSuperclassMethod(
+  Function meth, Function calledMulti, DataFlow::MethodCallNode call1,
+  DataFlow::MethodCallNode call2, string name
+) {
+  exists(Class cls |
+    meth.getName() = name and
+    meth.getScope() = cls and
+    locationBefore(call1.getLocation(), call2.getLocation()) and
+    calledMulti = getASuperCallTargetFromCall(cls, meth, call1, name) and
+    calledMulti = getASuperCallTargetFromCall(cls, meth, call2, name) and
```

Comment:
Hmm... I guess if you have $n$ calls to a superclass method, then this will produce $n\cdot(n-1)/2$ results, though I guess this might be rare in general.

As an alternative, we could consider `rank`ing the superclass method calls by location, and picking the first two items in this ranking. That way, we only ever produce two method calls.